Core/Src/main.c

```diff
@@ -87,16 +89,37 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
+  /*CS pull up to end any ongoing transmissions*/
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, 0.05 * 64000); /*set compare register value to 5% duty cycle time*/
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1); /*start the timer*/
+  /* takes in 3 bytes of data format found in figures 6-1 and 6-2*/
+  uint8_t data_transmit[3] = {1, 128, 0}; /*This connects to channel 0 where D2 D1 D0 are x00 respectively*/
+  uint8_t data_receive[3] = {0, 0, 0}; /*set up received bits array*/
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
-    /* USER CODE END WHILE */
+	HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); /*pull cs low to start transmission*/
+
+	HAL_SPI_TransmitReceive(&hspi1, data_transmit, data_receive, 3, HAL_MAX_DELAY); /*receive transmitted data from adc*/
+
+	HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); /*pull cs high to end transmission*/
+
+	uint16_t adc_relevant_val = ((data_receive[1] & 3) << 8) | data_receive[2]; /*Only retrieve the last 10 bits*/
 
+	float adc_config = (((adc_relevant_val/1023)*0.05)+0.05) *64000; /*Normalize these bits to be between 0.05 and 0.1 (the duty cycle)*/
```

Comment:
The `(adc_relevant_val/1023)` part of calculation will result in an undesired result here. Since non of `adc_relevant_val` or `1023` is a floating point number. The computer will execute a whole number division instead of a floating point division here.